### ISSUE TITLE
client-side navigation with a quote causes internal failure

### ISSUE THREAD

**Issue by chrisvariety** [brillout/vite-plugin-ssr#47]
```
[Error] Unhandled Promise Rejection: Error: [vite-plugin-ssr@0.1.0-beta.34][Internal Failure] You stumbled upon an bug in `vite-plugin-ssr`'s source code (an internal assertion failed). This should definitely not be happening, and you should create a new issue at https://github.com/brillout/vite-plugin-ssr/issues/new that includes this error stack (the error stack is enough to debug this). Or reach out on Discord. A fix will be written promptly.
	(anonymous function) (vite-plugin-ssr_client_router.js:1029)
	asyncFunctionResume
	(anonymous function)
	promiseReactionJobWithoutPromise
	promiseReactionJob
```

<img width="351" alt="Screenshot 2021-04-25 at 11 01 01@2x" src="https://user-images.githubusercontent.com/44855/115998509-8e4ce600-a5b5-11eb-85bb-f8ff8b8860ff.png">

Reproduction is something like:

```
const searchQuery = '"a"';
navigate(`/?q=${searchQuery}`);
```

Note changing it to:

```
navigate(`/?q=${encodeURIComponent(searchQuery)}`);
```
...works fine, so it may just be a matter of documentation / a better error message.

(this is a minor issue)

**Comment by brillout** [comment]
I will have a look at it later today.

Curious; what's your dev environment? You have an unexpected strack trace.

**Comment by chrisvariety** [comment]
I think that's just Safari, here's Chrome:

```
newError.ts:18 Uncaught (in promise) Error: [vite-plugin-ssr@0.1.0-beta.34][Internal Failure] You stumbled upon an bug in `vite-plugin-ssr`'s source code (an internal assertion failed). This should definitely not be happening, and you should create a new issue at https://github.com/brillout/vite-plugin-ssr/issues/new that includes this error stack (the error stack is enough to debug this). Or reach out on Discord. A fix will be written promptly.
    at newError2 (newError.ts:18)
    at assert (assert.ts:18)
    at parseUrl (parseUrl.ts:13)
    at getFileUrl (getFileUrl.ts:19)
    at retrieveContextProps (getContextProps.client.ts:22)
    at getContextProps (getContextProps.client.ts:16)
    at getPageByUrl.client.ts:14
    at getPageByUrl (getPageByUrl.client.ts:14)
    at fetchAndRender (useClientRouter.client.ts:71)
    at navigateFunction (useClientRouter.client.ts:45)
    at navigate (useClientRouter.client.ts:136)
    at onSubmit (Catalog.tsx:34)
    at HTMLUnknownElement.callCallback2 (react-dom.development.js:3945)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:3994)
    at invokeGuardedCallback (react-dom.development.js:4056)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:4070)
    at executeDispatch (react-dom.development.js:8243)
    at processDispatchQueueItemsInOrder (react-dom.development.js:8275)
    at processDispatchQueue (react-dom.development.js:8288)
    at dispatchEventsForPlugins (react-dom.development.js:8299)
    at react-dom.development.js:8508
    at batchedEventUpdates$1 (react-dom.development.js:22396)
    at batchedEventUpdates (react-dom.development.js:3745)
    at dispatchEventForPluginEventSystem (react-dom.development.js:8507)
    at attemptToDispatchEvent (react-dom.development.js:6005)
    at dispatchEvent (react-dom.development.js:5924)
    at unstable_runWithPriority (scheduler.development.js:468)
    at runWithPriority$1 (react-dom.development.js:11276)
    at discreteUpdates$1 (react-dom.development.js:22413)
    at discreteUpdates (react-dom.development.js:3756)
    at dispatchDiscreteEvent (react-dom.development.js:5889)
```

**Comment by brillout** [comment]
I see what's going on, I will fix it tomorrow.

**Comment by brillout** [comment]
Released in `0.1.0-beta.35`.

This required some profound changes; hopefully I didn't break other aspects of URL handling (but I'm confident I didn't :-)). Good news it that the new URL handling structure is fundamentally more robust. Happy to fix any future URL handling edge cases that may come up.

It may sound surprising but, on second thought, testing URL handling is actually mostly useless (happy to elaborate if you are curious) so I'm not going to programmatically test URL handling for the time being.
